plugins/vpp/natplugin/descriptor/nat44_vrf_route.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2018 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package descriptor
+
+import (
+	"go.ligato.io/cn-infra/v2/logging"
+	kvs "go.ligato.io/vpp-agent/v3/plugins/kvscheduler/api"
+	"go.ligato.io/vpp-agent/v3/plugins/vpp/natplugin/descriptor/adapter"
+	"go.ligato.io/vpp-agent/v3/plugins/vpp/natplugin/vppcalls"
+	nat "go.ligato.io/vpp-agent/v3/proto/ligato/vpp/nat"
+)
+
+const (
+	NAT44VrfRouteDescriptorName = "vpp-nat44-vrf-route"
+)
+
+type NAT44VrfRouteDescriptor struct {
+	log        logging.Logger
+	natHandler vppcalls.NatVppAPI
+
+	// UseDeprecatedAPI tracks whether deprecated global API (NAT interfaces, addresses) is being used on NB.
+	// Used to orchestrate which data should be dumped from which descriptor on Retrieve.
+	UseDeprecatedAPI bool
+}
+
+// NewNAT44VrfRouteDescriptor creates a new instance of the VrfRoute descriptor.
+func NewNAT44VrfRouteDescriptor(natHandler vppcalls.NatVppAPI, log logging.PluginLogger) *kvs.KVDescriptor {
+	ctx := &NAT44VrfRouteDescriptor{
+		natHandler: natHandler,
+		log:        log.NewLogger("nat44-vrf-route-descriptor"),
+	}
+	typedDescr := &adapter.NAT44VrfRouteDescriptor{
+		Name:                 NAT44VrfRouteDescriptorName,
+		NBKeyPrefix:          nat.ModelNat44VrfRoute.KeyPrefix(),
+		ValueTypeName:        nat.ModelNat44VrfRoute.ProtoName(),
+		KeySelector:          nat.ModelNat44VrfRoute.IsKeyValid,
+		KeyLabel:             nat.ModelNat44VrfRoute.StripKeyPrefix,
+		Create:               ctx.Create,
+		Delete:               ctx.Delete,
+		Dependencies:         ctx.Dependencies,
+		RetrieveDependencies: []string{NAT44VrfDescriptorName},
```

Comment:
Not sure about whether this one is needed as not retrieve is implemented in this descriptor.

Reply:
However, it might be implemented in the future. If it does not prevent normal function let's keep it. It helps to make sure the VRFs are always retrieved before routes, since the route might be dependent on some VRF and having it in reverse order would fail to resolve the VRF.